internal/controller/clusterdeployment_controller.go

```diff
@@ -388,13 +390,71 @@ func (r *ClusterDeploymentReconciler) updateCluster(ctx context.Context, mc *kcm
 	return ctrl.Result{}, nil
 }
 
+func (r *ClusterDeploymentReconciler) updateSveltosClusterCondition(ctx context.Context, clusterDeployment *kcm.ClusterDeployment) (bool, error) {
+	gvr := schema.GroupVersionResource{
+		Group:    libsveltosv1beta1.GroupVersion.Group,
+		Version:  libsveltosv1beta1.GroupVersion.Version,
+		Resource: "sveltosclusters",
+	}
+
+	sveltosClusters, err := r.DynamicClient.Resource(gvr).Namespace(clusterDeployment.Namespace).List(ctx, metav1.ListOptions{
```

Comment:
i do not fully understand, why cannot we utilize the typed client. it'd be much easier, IMO. are there any limitations?